src/recipe/Baked.Recipe.Service.Application/Lifetime/Scoped/ScopedLifetimeFeature.cs

```diff
@@ -11,34 +11,21 @@ public void Configure(LayerConfigurator configurator)
             builder.Index.Type.Add<ScopedAttribute>();
         });
 
-        configurator.ConfigureServiceCollection(services =>
-        {
-            services.AddFromAssembly(configurator.Context.GetGeneratedAssembly(nameof(ScopedLifetimeFeature)));
-        });
-
-        configurator.ConfigureGeneratedAssemblyCollection(generatedAssemblies =>
+        configurator.ConfigureDomainServicesModel(model =>
         {
             var domain = configurator.Context.GetDomainModel();
+            foreach (var scoped in domain.Types.Having<ScopedAttribute>())
+            {
+                model.Services.AddScoped(scoped, useFactory: true);
 
-            generatedAssemblies.Add(nameof(ScopedLifetimeFeature),
-                assembly =>
-                {
-                    assembly
-                        .AddReferenceFrom<ScopedLifetimeFeature>()
-                        .AddCodes(new ScopedServiceAdderTemplate(domain));
+                scoped.Apply(t => model.References.Add(t.Assembly));
```

Comment:
not sure why this is needed? why not add all domain assemblies to service model references directly from domain assemblies business feature?